### PR TITLE
refactor: Refactor FPE masks in Examples

### PR DESCRIPTION
**Pull request by andiwand**
We are currently mixing where we set masks and some of them are not accurate anymore. Here I try to reduce the masks to a minimum and set them in the code when appropriate.

blocked by:
- https://github.com/acts-project/acts/pull/2580

### CODE REVIEW

File: CI/physmon/workflows/physmon_track_finding_ttbar.py [hunk 1]
```diff
@@ -34,57 +33,10 @@
...
-            "Fatras/include/ActsFatras/Kernel/Simulation.hpp",
-            (96, 97),
-            acts.FpeType.FLTOVF,
-            1,
-        ),
-    ]
     s = acts.examples.Sequencer(
         events=3,
-        numThreads=-1,
+        numThreads=1,
```
Comment (paulgessinger) [inline CI/physmon/workflows/physmon_track_finding_ttbar.py:38]: Intentional?

File: Core/include/Acts/TrackFitting/detail/GsfUtils.hpp [hunk 2]
```diff
@@ -194,7 +194,9 @@ void computePosteriorWeights(
             .data(),
         state.predictedCovariance(), state.projector(), state.calibratedSize());
 
+    // MARK: fpeMaskBegin(FLTUND, 1, #2589)
     const auto factor = std::sqrt(1. / detR) * std::exp(-0.5 * chi2);
```
Comment (paulgessinger) [inline Core/include/Acts/TrackFitting/detail/GsfUtils.hpp:198]: Would the `safeExp` function help here?

File: Core/include/Acts/Vertexing/AdaptiveMultiVertexFinder.ipp [hunk 3]
```diff
@@ -116,8 +116,10 @@ auto Acts::AdaptiveMultiVertexFinder<vfitter_t, sfinder_t>::find(
         break;
       }
     }
+    // MARK: fpeMaskBegin(FLTUND, 1, #2590)
     bool keepVertex = isGoodVertex &&
                       keepNewVertex(vtxCandidate, allVerticesPtr, fitterState);
```
Comment (paulgessinger) [inline Core/include/Acts/Vertexing/AdaptiveMultiVertexFinder.ipp:121]: Could the mask go further in to `keepNewVertex`?

Comment (andiwand) [reply Core/include/Acts/Vertexing/AdaptiveMultiVertexFinder.ipp:121]: the stacktrace did not point me further down and I didn't want to guess

**Review by paulgessinger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [166c57a7a2883db2f452a738a3ddb39b312ee1d3]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6402607)!

|status|job|
|------|---|
|:red_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33503163) |
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502915) |
|:red_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502701) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502700) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502699) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502698) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502697) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502696) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502695) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502694) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502693) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502692) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502691) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502690) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502689) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502688) |
|:red_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502686) |
|:red_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33502685) |

**Pull request merged**